native/dgelq2.go

```diff
@@ -0,0 +1,44 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package native
+
+import "github.com/gonum/blas"
+
+// Dgelq2 computes the LQ factorization of the m×n matrix a.
+//
+// During Dgelq2, a is modified to contain the information to construct Q and L.
+// The lower triangle of a contains the matrix L. The upper triangular elements
+// (not including the diagonal) contain the elementary reflectors. Tau is modified
+// to contain the reflector scales. Tau must have length of at least k = min(m,n)
+// and will panic otherwise.
+//
+// See Dgeqr2 for a description of the elementary reflectors and orthonormal
+// matrix Q. Q is constructed as a product of these elementary reflectors,
+// Q = H(k) ... H(2) H(1).
+//
+// Work is a temporary storage slice of length at least m and will panic otherwise.
+func (impl Implementation) Dgelq2(m, n int, a []float64, lda int, tau, work []float64) {
```

Comment:
The data types (so far) are identical to BLAS. blas/native has a large doc comment about the data layout http://godoc.org/github.com/gonum/blas/native . There is a comment in the lapack package level documentation to this effect: "Slice function arguments frequently represent vectors and matrices. The data layout is identical to that found in https://godoc.org/github.com/gonum/blas/native. "
I don't think it's worth doubling the documentation on layout information. Users will hopefully be using mat64, and if not then hopefully at least blas64 and lapack64. If we can modify the lapack package documentation to be more clear, I'm all for it.